app/scripts/analysis/tools/powercalc/PowercalcMainCtrl.ctrl.coffee

```diff
@@ -0,0 +1,1529 @@
+      @TwoTGUI_n1 = Math.ceil(@TwoTGUI_n1)
+      @TwoTGUI_maxn2 = Math.max(@TwoTGUI_maxn1, @TwoTGUI_n1)
+    @TwoTGUI_checkRange()
+    @TwoTGUI_click()
+    @TwoTGUI_graph()
+    return
+  TwoTGUI_graph:() ->
+    @TwoTGUI.drawNormalCurve(@TwoTGUI_mean1, Math.pow(@TwoTGUI_sigma1, 2), @TwoTGUI_sigma1, @TwoTGUI_mean2, Math.pow(@TwoTGUI_sigma2, 2), @TwoTGUI_sigma2, @TwoTGUI_alpha);
+    if @deployed
+      $("#display_legend1").text(@comp_agents[0]+": "+@TwoTGUI_mean1.toFixed(3))
+      $("#display_legend2").text(@comp_agents[1]+": "+@TwoTGUI_mean2.toFixed(3))
+      $("#display_legend1").css("background-color","aquamarine")
+      $("#display_legend2").css("background-color","chocolate")
+    else
+      $("#display_legend1").text("Sample1: " + @TwoTGUI_mean1.toFixed(3))
+      $("#display_legend2").text("Sample2: " + @TwoTGUI_mean2.toFixed(3))
+      $("#display_legend1").css("background-color","aquamarine")
+      $("#display_legend2").css("background-color","chocolate")
+  TwoTGUI_changeValue: (evt) ->
+    name = evt.target.name
+    val = evt.target.value
+    key = evt.which or evt.keyCode
+    if name is "TwoTGUI_n1"
+      @TwoTGUI_n1 = parseFloat(val)
+    if name is "TwoTGUI_n2"
+      @TwoTGUI_n2 = parseFloat(val)
+    if name is "TwoTGUI_mean1"
+      @TwoTGUI_mean1 = parseFloat(val)
+    if name is "TwoTGUI_mean2"
+      @TwoTGUI_mean2 = parseFloat(val)
+    if name is "TwoTGUI_sigma1"
+      @TwoTGUI_sigma1 = parseFloat(val)
+    if name is "TwoTGUI_sigma2"
+      @TwoTGUI_sigma2 = parseFloat(val)
+    if name is "TwoTGUI_power"
+      @TwoTGUI_power = parseFloat(val)
+    if key is 13
+      if name is "TwoTGUI_power"
+        @TwoTGUI_powerTon()
+        return
+      @TwoTGUI_update()
+      return
+  TwoTGUI_show_help: () ->
+    if @TwoTGUI_help is true
+      $('#TwoTGUI_H').val "Show Help"
+    else
+      $('#TwoTGUI_H').val "Hide Help"
+    @TwoTGUI_help = !@TwoTGUI_help
+    return
+  TwoTGUI_t_test: () ->
+    v1 = Math.pow(@TwoTGUI_sigma1,2) / @TwoTGUI_n1
+    v2 = Math.pow(@TwoTGUI_sigma2,2) / @TwoTGUI_n2
+    df =  Math.pow((v1 + v2),2) / (Math.pow(v1,2) / (@TwoTGUI_n1 - 1.0) + Math.pow(v2,2) / (@TwoTGUI_n2 - 1.0))
+    @TwoTGUI_t = @tdistr(df, 1-@TwoTGUI_alpha)
+    @TwoTGUI_pvalue = @tprob(df, @TwoTGUI_t)
+  TwoTGUI_reset: () ->
+    @TwoTGUI_n1 = 10
+    @TwoTGUI_n2 = 10
+    @TwoTGUI_maxn1 = 20
+    @TwoTGUI_maxn2 = 20
+    @TwoTGUI_maxn = 20
+    @TwoTGUI_mean1 = 10
+    @TwoTGUI_mean2 = 10
+    @TwoTGUI_meanMax1 = 20
+    @TwoTGUI_meanMax2 = 20
+    @TwoTGUI_meanMax = 20
+    @TwoTGUI_sigma1 = 20
+    @TwoTGUI_sigma2 = 20
+    @TwoTGUI_sigmaMax1 = 40
+    @TwoTGUI_sigmaMax2 = 40
+    @TwoTGUI_sigmaMax = 40
+    @TwoTGUI_alpha = 0.010
+    @TwoTGUI_power = 0
+    @TwoTGUI_powerMax = 1 
+    @TwoTGUI_t = 0
+    @TwoTGUI_pvalue = 0
+    @TwoTGUI_mode = "Two Tailed"
+    @TwoTGUI_update()
+
+
```

Comment:
avoid leaving too much of empty space. leave a commented line of symbols if needed, e.g. `# ===============`